src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,219 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Parses given string to operate on.
+ *
+ * @author Aaron Tan
+ */
+public class Parser {
+
+    private TaskList tasks;
+
+    /**
+     * Processes String s to be operated on.
+     *
+     * @param s String to be processed.
+     * @param tasks Task list to be operated on.
+     */
+    protected void process(String s, TaskList tasks) {
+        this.tasks = tasks;
+        String[] words = s.split(" ");
+        String command = words[0];
+        switch (command) {
```

Comment:
I like your use of switch statements! That makes your program more readable compared to if-else statements!

Reply:
Agreed!